components/eamxx/src/dynamics/homme/atmosphere_dynamics.cpp

```diff
@@ -175,14 +167,14 @@ void HommeDynamics::set_grids (const std::shared_ptr<const GridsManager> grids_m
   add_field<Computed>("pseudo_density",FL({COL,     LEV},{ncols,  nlev_mid}),Pa,    pgn,N);
   add_field<Computed>("pseudo_density_dry",FL({COL, LEV},{ncols,  nlev_mid}),Pa,    pgn,N);
   add_field<Updated> ("ps",            FL({COL         },{ncols           }),Pa,    pgn);
-  add_field<Updated >("qv",            FL({COL,     LEV},{ncols,  nlev_mid}),Q,     pgn,"tracers",N);
-  add_field<Updated >("phis",          FL({COL         },{ncols           }),m2/s2, pgn);
+  add_field<Updated> ("qv",            FL({COL,     LEV},{ncols,  nlev_mid}),Q,     pgn,"tracers",N);
   add_field<Computed>("p_int",         FL({COL,    ILEV},{ncols,  nlev_int}),Pa,    pgn,N);
   add_field<Computed>("p_mid",         FL({COL,     LEV},{ncols,  nlev_mid}),Pa,    pgn,N);
   add_field<Computed>("p_dry_int",     FL({COL,    ILEV},{ncols,  nlev_int}),Pa,    pgn,N);
   add_field<Computed>("p_dry_mid",     FL({COL,     LEV},{ncols,  nlev_mid}),Pa,    pgn,N);
   add_field<Computed>("omega",         FL({COL,     LEV},{ncols,  nlev_mid}),Pa/s,  pgn,N);
-  add_group<Updated>("tracers",pgn,N, Bundling::Required);
+  add_group<Updated> ("tracers",pgn,N, Bundling::Required);
+  create_helper_field("phis",{COL},{ncols},pgn);
```

Comment:
Aren't we storing phis in the grids? Can't we just grab it from there, instead of creating another field?

Reply:
I believe I need to use fields because I want to use them in the `m_ic_remapper` in `initialize_homme_state()`.

Reply:
My opinion: phi_s should remain a field like ps, unless there was some crucial need that motivated this change. A grid object most naturally has as data the topology and geometry of the nodes and edges, and nothing else. In principle, multiple models could use it, only some of which know about phi_s.

Reply:
Ah, yes, we need to be able to write them, and the grid fields are read only. I have a commit in a follow up PR that allows to extract nonconst geo data from a const grid. In that PR, I can remove this helper field then.

Reply:
@bartgol I'm saying something a bit stronger than that. I think it's a design mistake to put phi_s in the grids class. What is the motivation for moving all these data there? As far as I can tell, it's not improving anything. In my opinion, if one just thinks of phi_s as ps, then it becomes clear where phi_s should live. As for the vertical coordinate, if you want to move it out of the dynamics AD object, then I suggest an object separate from "grid". Note that, e.g., a pg2 and GLL grid both have the same reference vertical coordinate, implying the vertical coordinate data are conceptually separate from the horizontal grids.

Reply:
@ambrad
> If this PR were to do only 2, would there be a lot fewer code changes?
 
Yes there would be a lot less changes, but even more duplicated code. The grid seemed like a common place to gather these things (lat/lon and vert coords were already stored as geometry data), but I guess it fell short of truly getting rid of almost all code duplication, which can easily be done. 

As far as keeping `phis` as a FM field, that is fine with me. There is no reason it can't be, I just had it as geometry data since it fit in with other geometry data lat/lon and vert coords in that it is loaded from file in the same way, but I see your point that it is not really grid data like lat/lon (though I don't quite understand why the vertical coords are not?).

I do think a separate PR needs to be done to remove the code duplication in loading lat/lon and vert coords from file, but that does not need to be done here. Maybe a separate utility function that gets a filename and list of fields and hands back the loaded data. Shouldn't that already exist for IC? Anyways, I'll trim this PR back to only be loading topo from file.